plugins/scaffolder/src/components/fields/MyGroupsPicker/MyGroupsPicker.tsx

```diff
@@ -82,10 +79,11 @@ export const MyGroupsPicker = (props: MyGroupsPickerProps) => {
     _: React.ChangeEvent<{}>,
     value: { label: string; ref: string } | null,
   ) => {
-    setSelectedGroup(value);
     onChange(value?.ref ?? '');
   };
 
+  const selectedEntity = groups?.find(e => e.ref === formData) || null;
```

Comment:
I think that we should put this in a `useEffect` for when the `groups` change, so that we can set the selected group if there's `formData`.

Something like:


```ts
useEffect(() => { 
  if (formData && !selectedGroup) {
    setSelectedGroup(groups?.find(e => e.ref === formData)
  }
}, [groups, formData, selectedGroup, setSelectedGroup])
```

Reply:
@Ali-Dalal the support both actually, it's how the majority of the other components work I believe.

Reply:
@benjdlambert I tried the suggestion on my local set up. it gives the same result (I need to add null check to avoid react rendering error).
but my question is, since `useAsync` (line 53) hook has no dependencies, it will get fired every time the component is mounted and it will also trigger `useEffect` hooks that have groups as a dependency. so will be there any value of adding the new `useEffect`?

Note: I checked other components (`EntityPicker`, `MultiEntityPicker`) and they all set the value without hooks.

Reply:
The change looks good to me! You always rely on formData and avoid glitches that occur when setting a state in an effect

Reply:
Could you please approve the PR?